compact_str/src/features/sqlx.rs

```diff
@@ -98,14 +99,17 @@ impl sqlx::postgres::PgHasArrayType for CompactString {
 #[cfg(feature = "sqlx-sqlite")]
 #[cfg_attr(docsrs, doc(cfg(feature = "sqlx-sqlite")))]
 impl<'q> Encode<'q, sqlx::Sqlite> for CompactString {
-    fn encode(self, buf: &mut <sqlx::Sqlite as HasArguments<'q>>::ArgumentBuffer) -> IsNull {
+    fn encode(
+        self,
+        buf: &mut <sqlx::Sqlite as Database>::ArgumentBuffer<'q>,
+    ) -> Result<IsNull, BoxDynError> {
         Encode::<'_, sqlx::Sqlite>::encode(self.into_string(), buf)
```

Comment:
```suggestion
        Encode::<'_, sqlx::Sqlite>::encode(self.as_str(), buf)
```

Needless allocation?

Reply:
I tried this change but it doesn't work. AFAICT what's happening is `buf` here is a [`SqliteArgumentValue`](https://docs.rs/sqlx/latest/sqlx/sqlite/enum.SqliteArgumentValue.html) which contains a `Cow<'q, str>`. Because we have `self` as an owned value, trying to use `self.as_str()` doesn't work because `self` gets dropped at the end of this scope, but needs to live for `'q`.

Reply:
Yes, the `'q` lifetime complicates life when Postgres or MySQL backend is used.